### DISCUSSION TITLE
Selecting everything

### DISCUSSION THREAD

**Discussion by ilyabo** [facebook/lexical#3635] [Q&A]
I am trying to implement something similar to Figma text boxes. There when a box is selected for editing (by double-clicking) all the text content gets selected:

<img width=300 src=https://user-images.githubusercontent.com/351828/209416008-1c8296d1-67b2-4f51-aab4-a437a605298d.png>

How can I tell Lexical to select everything in an editor?

**Comment by ilyabo** [comment]
After some trial and error I found this way of doing it:

```
editor.update(() => {
  const selection = $createRangeSelection();
  const node = $getRoot().getAllTextNodes()[0];
  if ($isTextNode(node)) {
    selection.focus.set(node.getKey(), 0, 'text');
    selection.anchor.set(node.getKey(), node.getTextContentSize(), 'text');
    $setSelection(selection);
  }
});
```

Note that it only works in case if you only have one text element, which is the case for me when I add a new node.

**Comment by longergrass** [comment]
This is a a simpler solution that worked for me:
```
const keyboardEvent = new KeyboardEvent("keydown", {});
editor.dispatchCommand(SELECT_ALL_COMMAND, keyboardEvent
```

Edit:
This is a better solution from the source code from:  https://github.dev/facebook/lexical/blob/7753924ca393027b8e04fa54c7624e4c2ee8a803/packages/lexical/src/LexicalUtils.ts#L1011-L1016

```
  const root = $getRoot();
  const selection = root.select(0, root.getChildrenSize());
  $setSelection($normalizeSelection(selection));
```
or $selectAll from @lexical/selection
```
 $selectAll()
```
